backend/src/services/GenerateReporter/GenerateSprintReporterService.ts

```diff
@@ -50,111 +49,101 @@ export class GenerateSprintReporterService {
-        this.calculateIterationBlockedAndDevelopingPercentage(
-          this.mapCardsByIteration(this.cards.matchedCards),
-          kanbanSetting.boardColumns
-        );
-      this.blockPercentage = this.sortBySprintStartDate(
-        sprintPercentageMap,
-        sprints
+      this.sprints = await kanban.getAllSprintsByBoardId(model);
+      const activeAndClosedSprints = this.getActiveAndClosedSprints(
+        this.sprints
+      );
+
+      const unorderedMapSprintBlockPercentage: Map<string, number> =
+        this.calculateBlockedAndDevelopingPercentage(mapSprintCards);
+      this.mapSprintBlockPercentage = this.sortBySprintStartDate(
+        unorderedMapSprintBlockPercentage,
+        activeAndClosedSprints
       );
 
-      const mapSprintStandardDeviation: Map<string, any> =
-        this.calculateStandardDeviation(this.cards);
-      this.standardDeviation = this.sortBySprintStartDate(
-        mapSprintStandardDeviation,
-        sprints
+      const unorderedMapSprintStandardDeviation: Map<string, number> =
+        this.calculateStandardDeviation(mapSprintCards);
+      this.mapSprintStandardDeviation = this.sortBySprintStartDate(
+        unorderedMapSprintStandardDeviation,
+        activeAndClosedSprints
       );
+
+      this.mapSprintBlockReasonPercentage = this.calculateBlockReasonPercentage(
+        activeAndClosedSprints,
+        mapSprintCards
+      );
+    }
+  }
+
+  private initTotalBlockTimeForEveryReasonMap(): Map<string, number> {
+    const totalBlockTimeForEveryReasonMap = new Map<string, number>();
+    for (let reason in JiraBlockReasonEnum) {
+      reason = Object.entries(JiraBlockReasonEnum).find(
+        ([key, val]) => key === reason
+      )?.[1]!;
+      totalBlockTimeForEveryReasonMap.set(reason, 0);
     }
+    return totalBlockTimeForEveryReasonMap;
   }
-  getLatestIterationSprintName(sprints: Sprint[]): string {
-    const sortedSprints = sprints
-      .filter((sprint) => sprint.startDate)
-      .sort((a, b) => Date.parse(a.startDate) - Date.parse(b.startDate));
+
+  private getLatestSprintName(sprints: Sprint[]): string {
+    const sortedSprints = sprints.sort(
+      (a, b) => Date.parse(a.startDate) - Date.parse(b.startDate)
+    );
     return sortedSprints.length > 0
       ? sortedSprints[sortedSprints.length - 1].name
       : "";
   }
 
-  getAllCardsByLatestSprintName(
-    cards: JiraCardResponse[],
-    sprintName: string
-  ): JiraCardResponse[] {
-    const allCardsInLatestSprint: JiraCardResponse[] = [];
-    cards.forEach((card) => {
-      if (card.baseInfo.fields.sprint === sprintName) {
-        allCardsInLatestSprint[allCardsInLatestSprint.length] = card;
-      }
-    });
-    return allCardsInLatestSprint;
-  }
-
   calculateBlockReasonPercentage(
-    cards: JiraCardResponse[],
-    boardColumns: RequestKanbanColumnSetting[] = [],
-    sprints: Sprint[]
+    sprints: Sprint[],
+    mapSprintCards: Map<string, JiraCardResponse[]>
```

Comment:
we only can get the data `mapSprintCards` in this class, because the method `mapCardsBySprintName` is private, but actually the method will be invoked by other method, and which need input the param `mapSprintCards`, so maybe need have some change about this param.